Fig7_PyGMT_datetime.py

```diff
@@ -0,0 +1,74 @@
+import requests
+import pandas as pd
+
+def get_star_history(repo):
+    owner = "GenericMappingTools"
+    headers = {
+        "Accept": "application/vnd.github.v3.star+json",
+        # "Authorization": "Bearer {YOUR_GITHUB_TOKEN_HERE}",
+    }
+
+    timestamps, users = [], []
+    page = 1
+
+    while True:
+        r = requests.get(
+            f"https://api.github.com/repos/{owner}/{repo}/stargazers",
+            headers=headers,
+            params={"per_page": 100, "page": page},
+        )
+        data = r.json()
+        if not data:
+            break
+
+        timestamps += [s["starred_at"] for s in data]  # full ISO 8601 timestamp
+        users += [s["user"]["login"] for s in data]
+        page += 1
+    df = pd.DataFrame({"timestamp": timestamps, "user": users}).sort_values("timestamp")
+
+    df.to_csv(f"star_history_github_{repo}.csv", sep=",", index=False)
+
+
+for repo in ["gmt", "pygmt", "gmt.jl", "gmtmex"]:
+    get_star_history(repo)
+
+# %%
+import datetime
+
+import pandas as pd
+import pygmt
+from pygmt.params import Box
+
+stars_step = 20
+
+fig = pygmt.Figure()
+fig.basemap(
+    projection="X12c/6c",
+    region=[datetime.date(2016, 1, 1), datetime.datetime.now(), -50, 1000],
+
+    frame=["x+lYear", "ya100f50+lGitHub stars"],

```

Comment:
```suggestion
    frame=["x", "ya100f50+lGitHub stars"],

```
I feel we can remove the "Year " x axis label since it's obvious that we are showing years here.